tests/tracer/test_sampler.py

```diff
@@ -646,7 +647,7 @@ def test_datadog_sampler_init():
         SamplingRule(sample_rate=0.5)
     ], "DatadogSampler initialized with default_sample_rate should hold a SamplingRule with that rate"
 
-    with override_env(dict(DD_TRACE_SAMPLE_RATE="0.5", DD_TRACE_RATE_LIMIT="10")):
+    with override_global_config(dict(_trace_sample_rate="0.5", _trace_rate_limit=10)):
```

Comment:
why are these test changes necessary?

Reply:
Previously the ``DD_TRACE_SAMPLE_RATE`` envar was always read when setting the sample rate. Now this environment variable is read once and it's value is stored in `config. _trace_sample_rate`. 

We need to override the global config object to run these tests.

Reply:
Ah I see, got it! We'd basically need to run these in a subprocess, initializing a new tracer each time in order for the envar to work.